gnucash/gnome/gnc-plugin-page-register.c

```diff
@@ -4603,10 +4609,42 @@ gnc_plugin_page_register_cmd_jump_associated_invoice (GtkAction* action,
     reg = gnc_ledger_display_get_split_register (priv->gsr->ledger);
     txn = gnc_split_register_get_current_trans (reg);
     invoice = invoice_from_split (gnc_split_register_get_current_split (reg));
-    if (xaccTransGetTxnType (txn) == TXN_TYPE_INVOICE)
-        invoice = invoice_from_split (xaccTransGetFirstAPARAcctSplit (txn, TRUE));
-    else
-        invoice = invoice_from_split (gnc_split_register_get_current_split (reg));
+
+    if (!invoice)
+    {
+        GList *invoices = invoices_from_transaction (txn);
+        if (!invoices)
+            PERR ("shouldn't happen");
+        else if (!invoices->next)
+            invoice = (GncInvoice*) invoices->data;
+        else
+        {
+            GList *details = NULL;
+            gint choice;
+            const gchar *amt;
+            for (GList *node = invoices; node; node = node->next)
+            {
+                GncInvoice* inv = (GncInvoice*) node->data;
+                amt = xaccPrintAmount
+                    (gncInvoiceGetTotal (inv),
+                     gnc_account_print_info (gncInvoiceGetPostedAcc(inv), TRUE));
+                details = g_list_prepend
+                    (details,
+                     g_strdup_printf ("Date=%s, ID=%s, Amount=%s",
+                                      qof_print_date (gncInvoiceGetDatePosted (inv)),
+                                      gncInvoiceGetID (inv), amt));
+            }
```

Comment:
This string must be translatable. Not sure the best string to describe an invoice.

Reply:
The description on the corresponding `GncEntry`?

Reply:
There is not necessarily a direct mapping between a GncEntry and a transaction split. Invoices can have multiple entries, which could be merged together or not when creating the invoice transaction.

IMO an invoice is well defined by a combination of date, owner, id and possibly total amount.

Reply:
So the dialog box should present Owner name, Invoice Number, Date, and Amount? How about whether it's a bill, invoice, or voucher as well? The user will need as much context as possible to work out which one they want to jump to.

Reply:
Candidate invoice description string in latest commit:

![image](https://user-images.githubusercontent.com/1975870/89405392-ffd4c380-d70a-11ea-968e-7aff0b75d5a0.png)

Reply:
> So the dialog box should present Owner name, Invoice Number, Date, and Amount? How about whether it's a bill, invoice, or voucher as well? The user will need as much context as possible to work out which one they want to jump to.

Good point. Add to that it can also be credit notes.

So any transaction can have a combination of these documented associated with it:
- bills and vendor credit notes
- invoices and customer credit notes
- employee vouchers and credit notes

Reply:
> Candidate invoice description string in latest commit:
> 
> ![image](https://user-images.githubusercontent.com/1975870/89405392-ffd4c380-d70a-11ea-968e-7aff0b75d5a0.png)

Looks good, except "Several invoices..." is incomplete as per my previous comment. It may be explicitly expanded to
"Several invoices and/or credit notes..." but I find that cumbersome.

Perhaps we should introduce a more generic term for these, like "customer documents", "vendor documents", "employee documents". That would still hold if someone in the future for example also implements pro-forma invoices or sales orders,...

Reply:
> So any transaction can have a combination of these documented associated with it:
> 
>     * bills and vendor credit notes
>     * invoices and customer credit notes
>     * employee vouchers and credit notes

`gncInvoiceGetTypeString` can return "Invoice" "Bill" "Expense" or "Credit Note". I _think_ it's adequate.

> Perhaps we should introduce a more generic term for these, like "customer documents", "vendor documents", "employee documents".

"business object"? "document"? Then the menu item in use "Open Associated Invoice" must also be modified.

Reply:
Resolved by renaming `invoice` to `document` in new strings. Existing menu-item `"Open Associated Invoice"` is unchanged.